include/aspect/initial_temperature/anelasticity_temperature.h

```diff
@@ -0,0 +1,145 @@
+/*
+  Copyright (C) 2016 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _aspect_initial_temperature_anelasticity_temperature_h
+#define _aspect_initial_temperature_anelasticity_temperature_h
+
+#include <aspect/initial_temperature/interface.h>
+#include <aspect/utilities.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    using namespace dealii;
+
+    /**
+     * A class that describes an initial temperature field for a 2D or 3D shear wave velocity (Vs) model.
+     * Vs values are converted to temperature using the anelasticity parameterization of Yamauchi & Takei (2016).
+     *
+     * @ingroup InitialTemperatures
+     */
+    template <int dim>
+    class AnelasticVs2T : public Utilities::AsciiDataInitial<dim>, public Interface<dim>
+    {
+      public:
+        /**
+        * Constructor. Initialize variables.
+        */
+        AnelasticVs2T ();
+
+        /**
+        * Initialization function. This function is called once at the
+        * beginning of the program. Checks preconditions.
+        */
+        void
+        initialize ();
+
+        // Avoid -Woverloaded-virtual:
+        using Utilities::AsciiDataInitial<dim>::initialize;
+
+        /**
+        * Return the boundary temperature as a function of position. For the
+        * current class, this function returns value from the text files.
+        */
+        double
+        initial_temperature (const Point<dim> &position) const;
+
+        /**
+        * Declare the parameters for the input files.
+        */
+        static
+        void
+        declare_parameters (ParameterHandler  &prm);
+
+        /**
+        * Read the parameters from the parameter file.
+        */
+        void
+        parse_parameters (ParameterHandler &prm);
+
+      private:
+
+        /**
+        * Function that assesses difference between input and calculated Vs for Brent minimization
+        */
+        double
+        fVs(double x, double depth, double absolute_Vs, double mu0, double dmudT,
```

Comment:
please make all parameters `const` to make sure they are only used for input

Reply:
Also unless you plan to use the functions outside of this class: since the functions are part of the class, you do not need to hand over member variables of this class to the function. You can access the member variables directly inside the function.

Reply:
I have now changed the function declaration so the input variables are no longer member variables. Was this what you had in mind?